public/pages/ConfigureForecastModel/components/AdvancedSettings/AdvancedSettings.tsx

```diff
@@ -0,0 +1,337 @@
+                ) {
+                  form.setFieldValue('imputationOption.custom_value', [
+                    { featureName: '', value: undefined },
+                  ]);
+                }
+              }, [field.value, form]);
+
+              return (
+                <>
+                  <FormattedFormRow
+                    fullWidth
+                    title="Sparse data handling"
+                    hint={[`Choose how to handle missing data points.`]}
+                    hintLink={FORECASTER_DOCS_LINK}
+                    isInvalid={isInvalid(field.name, form)}
+                    error={getError(field.name, form)}
+                  >
+                    <div style={{ maxWidth: FIELD_MAX_WIDTH }}>
+                      <EuiCompressedSelect 
+                        {...field} 
+                        options={sparseDataOptions}
+                        disabled={!isEditable}
+                      />
+                    </div>
+                  </FormattedFormRow>
+
+                  {field.value === SparseDataOptionValue.CUSTOM_VALUE && (
+                    <div style={{ maxWidth: FIELD_MAX_WIDTH }}>
+                      <EuiSpacer size="m" />
+                      <EuiText size="xs">
+                        <h5>Custom value</h5>
+                      </EuiText>
+                      <EuiSpacer size="s" />
+                      <FieldArray name="imputationOption.custom_value">
+                        {(arrayHelpers) => (
+                          <>
+                            {form.values.imputationOption.custom_value?.map(
+                              (_, index) => (
+                                <EuiFlexGroup
+                                  key={index}
+                                  gutterSize="s"
+                                  alignItems="center"
+                                >
+                                  <EuiFlexItem grow={false}>
+                                    <Field
+                                      name={`imputationOption.custom_value.${index}.data`}
+                                      id={`imputationOption.custom_value.${index}.data`}
+                                    >
+                                      {({ field }: FieldProps) => (
+                                        <div style={{ width: INPUT_SLIDER_WIDTH }}>
+                                          <EuiCompressedFieldNumber
+                                            style={{ width: '100%' }}
+                                            placeholder="Custom value"
+                                            name={field.name}
+                                            disabled={!isEditable}
+                                            onChange={(e) => {
+                                              const raw = e.target.value;
+                                              form.setFieldValue(field.name, raw === '' ? '' : Number(raw));
+                                            }}
```

Comment:
nit: seeing repeat onChange blocks with identical logic multiple times in this file. Maybe consider to extract it out as a common method

Reply:
extracted a method:

```
const toNumberOrEmpty = (value: string): number | '' => {
  return value === '' ? '' : Number(value);
};
```